test/sqllogictest/timezone.slt

```diff
@@ -150,3 +165,67 @@ query T
 SELECT '0001-01-01 12:00:00 -6'::TIMESTAMPTZ AT TIME ZONE '-5';
 ----
 0001-01-01 23:00:00
+
+query T
+SELECT timezone_offset('America/New_York', '2023-01-01T00:00:00+00')
+----
+(EST,-05:00:00,00:00:00)
+
+query T
+SELECT timezone_offset('America/Los_Angeles', '2023-01-01T00:00:00+00')
+----
+(PST,-08:00:00,00:00:00)
+
+query T
+SELECT timezone_offset('america/new_york', '2023-01-01T00:00:00+00')
+----
+(EST,-05:00:00,00:00:00)
+
+query T
+SELECT timezone_offset('America/New_York', '2023-01-01T00:00:00+00').abbrev
+----
+EST
+
+query T
+SELECT timezone_offset('America/New_York', '2023-01-01T00:00:00+00').base_utc_offset
+----
+-05:00:00
+
+query T
+SELECT timezone_offset('America/New_York', '2023-01-01T00:00:00+00').dst_offset
+----
+00:00:00
+
+query T
+SELECT timezone_offset('America/New_York', '2023-06-01T00:00:00+00').base_utc_offset
+----
+-05:00:00
+
+query T
+SELECT timezone_offset('America/New_York', '2023-06-01T00:00:00+00').dst_offset
+----
+01:00:00
+
+query T
+SELECT timezone_offset('America/New_York', '2023-11-05T05:00:00+00')
+----
+(EDT,-05:00:00,01:00:00)
+
+query T
+SELECT timezone_offset('America/New_York', '2023-11-05T05:30:00+00')
+----
+(EDT,-05:00:00,01:00:00)
+
+query T
+SELECT timezone_offset('America/New_York', '2023-11-05T06:00:00+00')
+----
+(EST,-05:00:00,00:00:00)
+
+query error invalid IANA Time Zone Database identifier: 'America/New_Yorks'
+SELECT timezone_offset('America/New_Yorks', now())
+
+query error invalid IANA Time Zone Database identifier: 'PST'
+SELECT timezone_offset('PST', now())
+
+query error invalid IANA Time Zone Database identifier: '-05'
+SELECT timezone_offset('-05', now())
```

Comment:
Here is a patch with some further test cases:

```
--- a/test/sqllogictest/timezone.slt
+++ b/test/sqllogictest/timezone.slt
@@ -221,6 +221,54 @@ SELECT timezone_offset('America/New_York', '2023-11-05T06:00:00+00')
 ----
 (EST,-05:00:00,00:00:00)
 
+query T
+SELECT timezone_offset('Europe/Vienna', '2023-11-05T06:00:00 UTC')
+----
+(CET,01:00:00,00:00:00)
+
+query T
+SELECT timezone_offset('Europe/Vienna', '2023-11-05T06:00:00 CET')
+----
+(CET,01:00:00,00:00:00)
+
+query T
+SELECT timezone_offset('Europe/Vienna', '2023-11-05T06:00:00 CEST')
+----
+(CET,01:00:00,00:00:00)
+
+query T
+SELECT timezone_offset('Europe/Vienna', '2023-06-05T06:00:00 CET')
+----
+(CEST,01:00:00,01:00:00)
+
+query T
+SELECT timezone_offset('Europe/Vienna', '2023-06-05T06:00:00 CEST')
+----
+(CEST,01:00:00,01:00:00)
+
+query T
+SELECT timezone_offset('Europe/Vienna', '2023-11-05T06:00:00')
+----
+(CET,01:00:00,00:00:00)
+
+query T
+SELECT timezone_offset('Europe/Vienna', '2023-11-05')
+----
+(CET,01:00:00,00:00:00)
+
+query T
+SELECT timezone_offset(NULL, '2023-11-05T06:00:00+00')
+----
+NULL
+
+query T
+SELECT timezone_offset('Europe/Vienna', NULL)
+----
+NULL
+
+query error invalid IANA Time Zone Database identifier: ''
+SELECT timezone_offset('', '2023-11-05T06:00:00+00')
+
 query error invalid IANA Time Zone Database identifier: 'America/New_Yorks'
 SELECT timezone_offset('America/New_Yorks', now())
 
-- 
```

I was surprised by the behavior of CEST (= CET + DST) behaving like CET. Is this correct?

Reply:
Thanks for the additional test cases, @nrainer-materialize! Would you mind just pushing them directly at this branch?

Reply:
I can try again, but I had issues pushing to your fork before.

Reply:
```
To github.com:benesch/materialize.git
 ! [remote rejected]       HEAD -> benesch/timezone-abbrevs-ii (permission denied)
error: failed to push some refs to 'github.com:benesch/materialize.git'
```

Reply:
> In both cases, you're asking "what's the offset for the Europe/Vienna timezone in the middle of the summer?" and the answer is the same: +1 from UTC, +1 for summer time, therefore abbreviation is CEST.

I was actually more surprise by the other test case in the winter (at `2023-11-05T06:00:00`). `CEST` behaves like `CET` (which is more surprising than `CET` adjusting to DST). But I guess it is okay.

Reply:
Hrm, that's very strange re: pushing to my fork.

<img width="312" alt="image" src="https://github.com/MaterializeInc/materialize/assets/882976/ae949abe-ee2f-49ee-8715-432fd320e7ca">

You're definitely using the right transport/auth for your `benesch` remote?

Reply:
Ok, found it. My local branch name is `benesch/timezone-abbrevs-ii`, therefore I need to execute `git push benesch HEAD:timezone-abbrevs-ii`.

Reply:
> Ok, found it. My local branch name is `benesch/timezone-abbrevs-ii`, therefore I need to execute `git push benesch HEAD:timezone-abbrevs-ii`.

Hooray!

> > In both cases, you're asking "what's the offset for the Europe/Vienna timezone in the middle of the summer?" and the answer is the same: +1 from UTC, +1 for summer time, therefore abbreviation is CEST.
> 
> I was actually more surprised by the other test case in the winter (at `2023-11-05T06:00:00`). `CEST` behaves like `CET` (which is more surprising than `CET` adjusting to DST). But I guess it is okay.

Ah, yeah, same thing there, though! 11/5 is well after the transition from summer time to standard time in the Europe/Vienna timezone. So you're just describing two different moments on 11/5 and asking what the current UTC offset for the Europe/Vienna timezone is, and in both cases it's "standard time, no summer time, therefore abbreviation is CET."

For sanity, outside of tests, I recommend always calling `timezone_offset` with a `timestamptz` whose timezone is UTC.